### PR TITLE
Backlog 11783

### PR DESCRIPTION
**Pull request by tdraier**
## JIRA

https://jira.jahia.org/browse/BACKLOG-11783

## Description

Use auto to publish packages. 
The script jahia-publish has been updated to rely on https://intuit.github.io/auto/home.html 
What it does : 
- Check that we are not master (release not allowed on branches for now)
- Detects the version number. Only commit affecting the current sub project are taken into account, thanks to the custom auto plugin. Skip release if there's no commit.
- Generates the changelog
- Update version number
- Commits and pushes
- Deploy to npm
- Create a github release and tag

## Tests

The following are included in this PR
- [ ] Unit Tests (Most changes _should_ have unit tests)
- [ ] Integration Tests

## Checklist

I have considered the following implications of my change: 

- [ ] Security (in particular for changes to authentication, authorization, data fetching, ...)
- [ ] Performance
- [ ] Migration
- [ ] Code maintainability

## Documentation

- [ ] Inline documentation
- [ ] Internal Documentation (wiki)
- [ ] User-facing Documentation

### CODE REVIEW

File: packages/scripts/publish.js [hunk 1]
```diff
@@ -1,48 +1,88 @@
...
+if (branchName !== 'master') {
+    console.log('Can only release from master branch');
+    process.exit(1);
 }
 
-let name = 'beta';
-if (branchName.startsWith('feature-')) {
-    name = branchName.replace(/-/g, '');
-    params.push('--tag', name);
+// Handle canary branches
```
Comment (FBerthelot) [inline packages/scripts/publish.js]: supress it :)

File: packages/scripts/publish.js [hunk 2]
```diff
@@ -1,48 +1,88 @@
 #!/usr/bin/env node
 const path = require('path');
 const fs = require('fs');
-const yarn = require('yarn-api');
 const branch = require('git-branch');
-const util = require('util');
+const spawn = require('cross-spawn');
+const dotenv = require('dotenv');
```
Comment (FBerthelot) [inline packages/scripts/publish.js]: no .env file is commited, is it normal ?

File: packages/scripts/auto-include-plugin.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        };
+    }
+
+    /** Tap into auto plugin points. */
+    apply(auto) {
+        auto.hooks.onCreateLogParse.tap(this.name, logParse => {
+            logParse.hooks.omitCommit.tap(this.name, commit => {
+                // Remove current working directory which is added by a path.resolve
+                // https://github.com/intuit/auto/blob/v8.6.0/packages/core/src/git.ts:309
+                let files = commit.files.map(f => f.substr(process.cwd().length + 1));
```
Comment (FBerthelot) [inline packages/scripts/auto-include-plugin.js]:
```suggestion
                const files = commit.files
                    .map(f => f.substr(process.cwd().length + 1))
                    .filter(f => this.options.paths.filter(p => f.startsWith(p)).length > 0); // Keep only files in the requested folder
```

File: packages/scripts/publish.js (hunk 2 above)
Comment (tdraier) [reply packages/scripts/publish.js]: Sure, it contains my personal github and npm keys :-)
I’ve added it to gitignore, btw

File: packages/scripts/publish.js (hunk 1 above)
Comment (tdraier) [reply packages/scripts/publish.js]: I still want to do it !

File: packages/scripts/publish.js (hunk 2 above)
Comment (DameniMilo) [reply packages/scripts/publish.js]: Could you add a .env.example file with the key that you need inside in order for this to work?
So one can just create a copy and set the parameters.

**Pull request merged**